libminifi/include/core/TypedValues.h

```diff
@@ -141,6 +141,47 @@ class TimePeriodValue : public TransformableValue, public state::response::UInt6
  * format <numeric> <byte size>.
  */
 class DataSizeValue : public TransformableValue, public state::response::UInt64Value {
+private:
+
+  //Signed
+  template<typename T, typename std::enable_if<std::is_signed<T>::value, T>::type* = nullptr>
+  static bool AssignIfFits(int64_t base, uint64_t mult, T& output) {
+    int64_t val = base * mult;
+    if(val >= std::numeric_limits<T>::min() && val <= std::numeric_limits<T>::max()) {
+      output = val;
+      return true;
+    }
+    return false;
+  }
+
```

Comment:
Does this code compile in windows? I'm not sure that it will.

Reply:
Appveyor showed that it didn't. Do you have access to windows? I can probably help with that since I do.

Reply:
Yes, please! I can check Appveyor err messages, but that's not so efficient in longer term.